### PR TITLE
H-4198: Remove `apps/hash` directory

### PR DESCRIPTION
**Pull request by vilkinsons**
## 🌟 What is the purpose of this PR?

We had previously kept `apps/hash` around so that external inbound links from offsite resources didn't break. With fewer of these links now in existence, we're removing this directory.

The contents of the `README.md` file that was present have been incorporated elsewhere into the repo, and various drive-by improvements to in-repo documentation have been made, ahead of our planned announcements.

Further improvements to the docs should be looked at.

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+localStorage["hash.internal.debugging"] = "true";
+```
+
+## Tests
+
+### Backend Integration Tests
+
+Backend integration tests are located in the [`/tests/hash-backend-integration`](/tests/hash-backend-integration) folder.
+
+_The tests require a running instance of `hash-external-services`. see [the root README](/README.md#external-services-test-mode) for information on doing this without polluting the development database._
```
Comment (TimDiekmann) [inline tests/README.md:23]: All tests require to run external services, this is the main reason why they are here and makes them integration tests in the first place. (Some benchmarks requires that as well, this is why they are here as well)

**Review by TimDiekmann — APPROVED** [review]
I think it's good to have the `apps/hash` folder to be removed. Also, I noticed that quite some documentation is outdated. I would propose to keep the outdated docs around for a bit longer but in the correct location, then we can improve it from there as otherwise (as I currently add quite a lot of docs) this will result in annoying merge conflicts.

I left some comments mainly for the future me.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -16,32 +16,226 @@ If you'd like to make a significant change or re-architecture to this repository
...
-To ascertain the license and contributing policy for any given project, check out the `LICENSE.md` and `CONTRIBUTING.md` files in its root.
+### Licenses
+
+To ascertain the license and contributing policy for any given project, check out the `LICENSE.md` and `CONTRIBUTING.md` files in its root (or the license information in the file itself, which takes precedence where present).
+
+### Tests
+
+All externally-contributed pull requests which modify code should be covered by one or more unit tests, at minimum.
+
+Please see the [`/tests` README](/tests/README.md) for more information about testing in HASH.
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply .github/CONTRIBUTING.md:33]: Further, we should have global instructions how to run tests and package-related instructions if it makes sense (this is done partially already)

File: apps/hash-graph/README.md [hunk 3]
```diff
@@ -94,7 +100,8 @@ To locate the build directory, run with `cargo build -vv` and search for "Genera
 
 New payloads can then be added in the `definitions` and `oneOf` of the `Status` schema.
 
-> Note: migrating to the above is in progress, and not all error responses currently satisfy the `Status` shape, consult the API spec to see.
+> [!NOTE]
+> Migrating to the above is in progress, and not all error responses currently satisfy the `Status` shape, consult the API spec to see.
```
[the remark this answers is not in the document]
Comment (vilkinsons) [reply apps/hash-graph/README.md:104]: Feel free to push directly to the branch, in fact.

File: apps/hash-graph/README.md [hunk 4]
```diff
@@ -14,11 +14,17 @@ This crate mainly creates a CLI interface to the graph crates defined in the [`l
...
+<!-- markdownlint-disable no-blanks-blockquote -->
+
+> [!IMPORTANT]  
+> At the moment, the _Graph_ starts the services it depends on differently to the rest of the codebase.
+>
+> **Before running the following command:**
 >
-> **Before running the following command, ensure you tear down any existing `external-services` that were started as outlined in the [README for the HASH application](/apps/hash/README.md).** Similarly, **ensure you call `deployment-down` before trying to run the `external-services`.**
+> 1. Ensure you tear down any existing `external-services` that were started as outlined in the ["Getting Started"](/README.md#--getting-started) section of the root README.
+> 2. Ensure you call `deployment-down` before trying to run the `external-services`.
```
[the remark this answers is not in the document]
Comment (vilkinsons) [reply apps/hash-graph/README.md:25]:
```suggestion
> - Ensure you tear down any existing `external-services` that were started as outlined in the ["Getting Started"](/README.md#--getting-started) section of the root README.
> - Ensure you call `deployment-down` before trying to run the `external-services`.
```

File: .github/CONTRIBUTING.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (vilkinsons) [reply .github/CONTRIBUTING.md:33]:
```suggestion
### Tests

All externally-contributed pull requests which modify code should, at minimum, be covered by one or more unit tests (colocated alongside the code). Other kinds of tests, including integration tests (found in the [`/tests` directory](/tests)) may also be required.

Please see the [`/tests` README](/tests/README.md) for more information about testing in HASH.
```

File: tests/README.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
+# Testing
+
+## Overview
+
+The `/tests` directory contains HASH tests which for one reason or another aren't inlined with the code they correspond to (e.g. unit tests). This includes end-to-end tests, integration tests, and load tests.
```
[the remark this answers is not in the document]
Comment (vilkinsons) [reply tests/README.md:5]:
```suggestion
The `/tests` directory contains HASH tests which for one reason or another aren't inlined with the code they correspond to (e.g. unit tests). This includes end-to-end tests, integration tests, performance benchmarks, and load tests.
```

File: tests/README.md (hunk 1 above)
Comment (vilkinsons) [reply tests/README.md:23]: Perhaps you could help in rewriting this section for clarity.

**Review by TimDiekmann — APPROVED** [review]
Thank you!
I created the follow-up tickets H-4208 and H-4209.
